sturdy/validator/apy_binning.py

```diff
@@ -81,7 +81,8 @@ def calculate_base_rewards(bins: dict[int, list[str]], miner_uids: list[str]) ->
             idx = miner_uids.index(uid)  # Get index directly from list
             base_rewards[idx] = base_reward
 
-    return base_rewards
+    # Normalize base rewards - otherwise we'll have negative values
+    return normalize_rewards(base_rewards)
```

Comment:
This doesn't fix negative rewards. See:

```
base_rewards = np.array([1, 0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0.0, -0.1])
max_reward = np.max(base_rewards)
rewards = base_rewards / max_reward
print(rewards)
[ 1.   0.9  0.8  0.7  0.6  0.5  0.4  0.3  0.2  0.1  0.  -0.1]
```

Reply:
This would be better as you already suggested in the discord: base_reward = max(0.0, 1.0 - (bin_idx * 0.1)